py/mphal.h

```diff
@@ -66,4 +66,11 @@ mp_uint_t mp_hal_ticks_ms(void);
 mp_uint_t mp_hal_ticks_us(void);
 #endif
 
+#ifndef mp_hal_pin_obj_t
+#define mp_hal_pin_obj_t mp_obj_t
+#define mp_hal_get_pin_obj(pin) (pin)
```

Comment:
Might be better to validate the pin object here and return a concrete pointer.  Eg:
```
#define mp_hal_pin_obj_t mp_obj_base_t*
mp_obj_base_t *mp_hal_get_pin_obj(mp_obj_t pin) {
    if (mp_obj_get_type(pin) != &mp_type_pin) {
        nlr_raise(..., "need a pin");
    }
    return MP_OBJ_TO_PTR(pin);
}
```
Then read/write can be inline, eg:
```
#define mp_hal_pin_read(pin) ((mp_pin_p_t*))pin->type->stream_p)->ioctl(pin, MP_PIN_READ, 0, NULL);
```